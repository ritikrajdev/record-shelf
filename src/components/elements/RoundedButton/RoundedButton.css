.rounded-button {
  outline: none;
  border: none;
  padding: 1rem 3rem;
  background-color: var(--light-purple);
  color: var(--text-color-secondary);

  font-weight: bold;
  font-size: 1.5rem;

  width: 250px;
  border-radius: 1.75rem;
}
