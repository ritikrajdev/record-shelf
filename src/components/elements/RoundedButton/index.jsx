import PropTypes from 'prop-types';
import React from 'react';

import './RoundedButton.css';

export default function RoundedButton({ children, onClick, style }) {
  return (
    <button className='rounded-button' style={style} onClick={onClick}>
      {children}
    </button>
  );
}

RoundedButton.propTypes = {
  children: PropTypes.node,
  onClick: PropTypes.func,
  style: PropTypes.object,
};

RoundedButton.defaultProps = {
  children: '',
};
