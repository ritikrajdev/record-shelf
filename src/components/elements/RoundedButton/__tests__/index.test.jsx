import { fireEvent, render } from '@testing-library/react';
import React from 'react';
import RoundedButton from '..';

describe('RoundedButton', () => {
  it('should render correctly', () => {
    const screen = render(<RoundedButton>Click me</RoundedButton>);
    expect(screen.getByText('Click me')).toBeTruthy();
    expect(screen.asFragment()).toMatchSnapshot();
  });

  it('should render correctly with no input', () => {
    const screen = render(<RoundedButton />);
    expect(screen.queryByText('Click me')).toBeFalsy();
    expect(screen.asFragment()).toMatchSnapshot();
  });

  it('should render correctly with style', () => {
    const screen = render(
      <RoundedButton style={{ backgroundColor: 'red' }}>Click me</RoundedButton>
    );
    expect(screen.getByText('Click me')).toBeTruthy();
    expect(screen.asFragment()).toMatchSnapshot();
  });

  it('should call onClick when clicked', () => {
    const onClick = jest.fn();
    const screen = render(
      <RoundedButton onClick={onClick}>Click me</RoundedButton>
    );
    const button = screen.getByText('Click me');
    fireEvent.click(button);
    expect(onClick).toHaveBeenCalled();
  });
});
