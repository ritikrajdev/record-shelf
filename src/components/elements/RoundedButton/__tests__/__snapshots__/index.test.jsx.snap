// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoundedButton should render correctly 1`] = `
<DocumentFragment>
  <button
    class="rounded-button"
  >
    Click me
  </button>
</DocumentFragment>
`;

exports[`RoundedButton should render correctly with no input 1`] = `
<DocumentFragment>
  <button
    class="rounded-button"
  />
</DocumentFragment>
`;

exports[`RoundedButton should render correctly with style 1`] = `
<DocumentFragment>
  <button
    class="rounded-button"
    style="background-color: red;"
  >
    Click me
  </button>
</DocumentFragment>
`;
